sdk/keyvault/keyvault-keys/test/tracing.test.ts

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { KeyClient, JsonWebKey } from "../src";
+import * as assert from "assert";
+import { TestTracer, setTracer, OperationTracingOptions } from "@azure/core-tracing";
+import { Context as MochaTestContext, Test } from "mocha";
+import { env, Recorder, isRecordMode, isPlaybackMode, delay } from "@azure/test-utils-recorder";
+import { authenticate } from "./utils/testAuthentication";
+import TestClient from "./utils/testClient";
+import { testPollerProperties } from "./utils/recorderUtils";
+import { isNode } from "@azure/core-http";
+
+// This test file confirms that all of our methods that create tracing spans
+// are effectively doing so.
+
+/**
+ * A catch-all type for the SpanGraph root and its nodes.
+ */
+interface TestRecursiveGraph {
```

Comment:
Just to do a quick one-off recursive walk to get all the names.

I guess that instead of doing this, what I really want is some standard way of extracting only the relevant information out of the tracing graph, so that I could test that information instead of having to know how to write the whole graph. Something similar to "get all the names". If this make sense, we could talk about the design and I could submit a PR to tracing before I go back to this pr 🌞

Reply:
I guess the reason I'm nervous about the "just check the names are right" approach is it's very easy to mis-parent spans when we forget to pass span contexts internally, and such a mistake wouldn't be caught by merely checking that all the right names exist.

You could maybe work around it a little bit by asserting there is only one root span, though.

Reply:
also there are cases where the same span might be included in two different places, but if you de-dupe the names you might miss that

Reply:
same *name* span, different instance, that is